### PR TITLE
utils.sh refactoring

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
1. Relocated a few functions to their caller script only, only if it is called by one parent script.
2. Logging updates
  a) Updated the contexts
  b) Change to MsgErr, if error or warning
3. Fixed the indentation
4. Re-wrote conditional statement with case statement

### CODE REVIEW

**Comment by juhlee-microsoft** [comment]
@LiliDeng @abhimarathe @lubaihua33 This is a big PR I ever made. Because of the requirement nature, I don't have any choice. I am running many tests for the change validation. If you review the PR earlier, it would be easy for us to close this PR before the end of this month. (End of 2019)

**Comment by juhlee-microsoft** [comment]
image

File: Testscripts/Linux/TIMESYNC-NTP.sh [hunk 1]
```diff
@@ -28,173 +28,161 @@ loopbackIP="127.0.0.1"              # IP to force NTPD to listen on IPv4
...
+            if [[ $os_RELEASE =~ 8.* ]]; then
+                LogMsg "Info: $os_VENDOR $os_RELEASE does not support NTP. Test skipped. "
+                SetTestStateSkipped
+                exit 0
+            fi
+            # Check if ntpd is running
+            if ! service ntpd restart
+            then
+                LogMsg "Info: NTPD not installed. Trying to install..."
+                if [ ! yum install -y ntp ntpdate ]; then
```
Comment (LiliDeng) [inline Testscripts/Linux/TIMESYNC-NTP.sh]: add [] will hit issue, that's why you see test fail on ubuntu

File: Testscripts/Linux/utils.sh [hunk 2]
```diff
@@ -1387,25 +1362,24 @@ CreateIfupConfigFile()
...
-				fi
+				if [ grep -q "$__interface_name" $__file_path ]; then
+                    LogErr "Warning will delete older configuration of interface $__interface_name"
+                    lineNumber=$(cat -n $__file_path |grep "iface $__interface_name"| awk '{print $1;}')
+                    if [ $lineNumber ]; then
+                        lineNumber=$lineNumber+1
+                        sed -i "${lineNumber},+1 d" $__file_path
+                    fi
+                    sed -i "/$__interface_name/d" $__file_path
+                fi
```
Comment (LiliDeng) [inline Testscripts/Linux/utils.sh]: indent is inconsistent between 1374 - 1382

File: Testscripts/Linux/utils.sh [hunk 3]
```diff
@@ -1959,72 +1873,28 @@ UpgradeMinorKernel() {
 	return 0
 }
 
-VerifyIsEthtool()
-{
+function VerifyIsEthtool() {
+	# Should have "return" value: 0, if existed. Otherwise, 1.
```
Comment (LiliDeng) [inline Testscripts/Linux/utils.sh:1877]: indent

File: Testscripts/Linux/utils.sh [hunk 4]
```diff
@@ -3257,6 +3050,9 @@ function stop_firewall() {
             iptables -F
             iptables -X
             ;;
+		coreos)
+            LogMsg "No extra steps need here."
```
Comment (LiliDeng) [inline Testscripts/Linux/utils.sh]: indent

File: Testscripts/Linux/TIMESYNC-NTP.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/TIMESYNC-NTP.sh]: Fixed

File: Testscripts/Linux/utils.sh (hunk 2 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/utils.sh]: fixed

File: Testscripts/Linux/utils.sh (hunk 3 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/utils.sh:1877]: fixed

File: Testscripts/Linux/utils.sh (hunk 4 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/utils.sh]: fixed

**Comment by juhlee-microsoft** [comment]
[SUSE SLES 15 latest] [LISAv2 Test Results Summary]
[SUSE SLES 15 latest] Test Run On           : 12/19/2019 22:35:19
[SUSE SLES 15 latest] ARM Image Under Test  : SUSE : SLES : 15 : latest
[SUSE SLES 15 latest] Initial Kernel Version: 4.12.14-5.30-azure
[SUSE SLES 15 latest] Final Kernel Version  : 4.12.14-5.30-azure
[SUSE SLES 15 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[SUSE SLES 15 latest] Total Time (dd:hh:mm) : 0:0:8
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[SUSE SLES 15 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[SUSE SLES 15 latest]     1 CORE                 TIMESYNC-NTP                                                                      PASS                 5.21 
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest] 
[SUSE SLES 15 latest] Logs can be found at C:\LISAv2\JU62\TestResults\2019-19-12-22-35-14-8950

**Comment by juhlee-microsoft** [comment]
[RedHat RHEL 8 latest] [LISAv2 Test Results Summary]
[RedHat RHEL 8 latest] Test Run On           : 12/19/2019 22:05:28
[RedHat RHEL 8 latest] ARM Image Under Test  : RedHat : RHEL : 8 : latest
[RedHat RHEL 8 latest] Initial Kernel Version: 4.18.0-80.11.2.el8_0.x86_64
[RedHat RHEL 8 latest] Final Kernel Version  : 4.18.0-80.11.2.el8_0.x86_64
[RedHat RHEL 8 latest] Total Test Cases      : 1 (0 Passed, 0 Failed, 0 Aborted, 1 Skipped)
[RedHat RHEL 8 latest] Total Time (dd:hh:mm) : 0:0:3
[RedHat RHEL 8 latest] 
[RedHat RHEL 8 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[RedHat RHEL 8 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[RedHat RHEL 8 latest]     1 CORE                 TIMESYNC-NTP                                                                   SKIPPED                 0.47 
[RedHat RHEL 8 latest] 
[RedHat RHEL 8 latest] 
[RedHat RHEL 8 latest] Logs can be found at C:\LISAv2\BC60\TestResults\2019-19-12-22-05-24-4202

[Canonical UbuntuServer 16.04-LTS latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 16.04-LTS latest] Test Run On           : 12/19/2019 22:05:08
[Canonical UbuntuServer 16.04-LTS latest] ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
[Canonical UbuntuServer 16.04-LTS latest] Initial Kernel Version: 4.15.0-1064-azure
[Canonical UbuntuServer 16.04-LTS latest] Final Kernel Version  : 4.15.0-1064-azure
[Canonical UbuntuServer 16.04-LTS latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 16.04-LTS latest] Total Time (dd:hh:mm) : 0:0:5
[Canonical UbuntuServer 16.04-LTS latest] 
[Canonical UbuntuServer 16.04-LTS latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 16.04-LTS latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 16.04-LTS latest]     1 CORE                 TIMESYNC-NTP                                                                      PASS                 0.99 
[Canonical UbuntuServer 16.04-LTS latest] 
[Canonical UbuntuServer 16.04-LTS latest] 
[Canonical UbuntuServer 16.04-LTS latest] Logs can be found at C:\LISAv2\RE66-20191219220503\TestResults\2019-19-12-22-05-05-1001

[Oracle Oracle-Linux 7.6 latest] [LISAv2 Test Results Summary]
[Oracle Oracle-Linux 7.6 latest] Test Run On           : 12/19/2019 22:05:22
[Oracle Oracle-Linux 7.6 latest] ARM Image Under Test  : Oracle : Oracle-Linux : 7.6 : latest
[Oracle Oracle-Linux 7.6 latest] Initial Kernel Version: 4.14.35-1902.6.6.el7uek.x86_64
[Oracle Oracle-Linux 7.6 latest] Final Kernel Version  : 4.14.35-1902.6.6.el7uek.x86_64
[Oracle Oracle-Linux 7.6 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Oracle Oracle-Linux 7.6 latest] Total Time (dd:hh:mm) : 0:0:5
[Oracle Oracle-Linux 7.6 latest] 
[Oracle Oracle-Linux 7.6 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Oracle Oracle-Linux 7.6 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Oracle Oracle-Linux 7.6 latest]     1 CORE                 TIMESYNC-NTP                                                                      PASS                 1.24 
[Oracle Oracle-Linux 7.6 latest] 
[Oracle Oracle-Linux 7.6 latest] 
[Oracle Oracle-Linux 7.6 latest] Logs can be found at C:\LISAv2\BH42\TestResults\2019-19-12-22-05-19-1192

[Canonical UbuntuServer 19.04 latest] [LISAv2 Test Results Summary]
[Canonical UbuntuServer 19.04 latest] Test Run On           : 12/19/2019 22:05:12
[Canonical UbuntuServer 19.04 latest] ARM Image Under Test  : Canonical : UbuntuServer : 19.04 : latest
[Canonical UbuntuServer 19.04 latest] Initial Kernel Version: 5.0.0-1025-azure
[Canonical UbuntuServer 19.04 latest] Final Kernel Version  : 5.0.0-1025-azure
[Canonical UbuntuServer 19.04 latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Canonical UbuntuServer 19.04 latest] Total Time (dd:hh:mm) : 0:0:6
[Canonical UbuntuServer 19.04 latest] 
[Canonical UbuntuServer 19.04 latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Canonical UbuntuServer 19.04 latest] -------------------------------------------------------------------------------------------------------------------------------------------
[Canonical UbuntuServer 19.04 latest]     1 CORE                 TIMESYNC-NTP                                                                      PASS                  1.5 
[Canonical UbuntuServer 19.04 latest] 
[Canonical UbuntuServer 19.04 latest] 
[Canonical UbuntuServer 19.04 latest] Logs can be found at C:\LISAv2\DE63\TestResults\2019-19-12-22-05-08-8771

**Comment by juhlee-microsoft** [comment]
Test   Snapshot | CORE | SR-IOV
-- | -- | --
Ubuntu 18.04 | 21/0/0/1 | 12/0/1/1
CentOS 8.0 | 21/0/0/1 | 13/0/0/1
SUSE 12 SP4 | 20/0/0/2 | 14/0/1/1
Passed/Failed/Aborted/Skipped |   |

**Comment by juhlee-microsoft** [comment]
I think they are good result with this branch. Please review and sign-off, @LiliDeng , @lubaihua33  @abhimarathe

**Comment by juhlee-microsoft** [comment]
@LiliDeng anything else?

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**